hack/tests/scaffolding/scaffold-memcached.go

```diff
@@ -174,6 +179,11 @@ func main() {
 		log.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
 	}
 
+	log.Print("Checking API rule violations")
+	if strings.Contains(string(cmdOut), "API rule violation") {
+		log.Fatalf("Error: %v\nCommand Output: %s\n", "API rule violations :", string(cmdOut))
+	}
```

Comment:
I'm in full agreement with adding a test to verify that we are not generating types that would violate API rules. However, this seems like it belongs in a unit test for the code used in the `operator-sdk generate openapi` command.

Unit tests are _much_ faster than e2e tests and then can be more precise because we have access to Go types and return values, not just the stdout and stderr of the command, which can make this test brittle.

Can you add a TODO comment on this test to transition this to a unit test in `cmd/operator-sdk/internal/genutil`? This might become obsolete in the SDK codebase when we begin inheriting scaffolds from Kubebuilder, but we could also contribute a test like this upstream.

Reply:
Sorry looks like there was a miscommunication on where to put the comment. Can you move it from `cmd/operator-sdk/internal/genutil/openapi.go` to this file, just above the `log.Print("Checking API rule violations")`. Something like:
```
// TODO(camilamacedo86) Move this test to a unit test in 
// `cmd/operator-sdk/internal/genutil/`. Unit tests are
// faster and are run more often during development, so it
// would be an improvement to implement this test there.
```

Reply:
|o| It makes more sense. I found the suggestion little confuse but ok. It was my bad. Keep simple.